### PR TITLE
JEN-927 8.0

### PR DESCRIPTION
**Pull request by abychko**
merge 3 awk calls in local/test-binary
[+] added separate function to process .output files

### CODE REVIEW

File: local/test-binary [hunk 1]
```diff
@@ -67,9 +84,8 @@ if [[ "${DEFAULT_TESTING}" != "no" ]]; then
         --max-test-fail=0 \
         --suite-timeout=9999 \
         --testcase-timeout=${TESTCASE_TIMEOUT} \
-        | tee ${WORKDIR_ABS}/mtr.output \
-        || status=$?
-    awk 'BEGIN { print "<testsuite name=\"MySQL\">" } /^([[:alnum:]]+).*[\[] (retry-)?pass [\]](.*)/ { print "<testcase name=\""$3"\" time=\"" $8/1000 "\"/>" } /^(.*) .*\[ skipped \]/ { print "<testcase name=\""$3"\"><skipped/></testcase>" } /^(.*) .*\[ (retry-)?fail \]/ { print "<testcase name=\""$3"\"><failure/></testcase>" } /^MTR\47s internal check of the test case \47(.*)\47 failed/ { print "<testcase name=\"testcase-check-"$8"\"><failure/></testcase>" } END { print "</testsuite>" }' < ${WORKDIR_ABS}/mtr.output > $WORKDIR_ABS/junit.xml
+        | tee ${WORKDIR_ABS}/mtr.output || status=$?
+    process_mtr_output MySQL ${WORKDIR_ABS}/mtr.output $WORKDIR_ABS/junit.xml
```
Comment (laurynas-biveinis) [inline local/test-binary]: Do args need quoting in case there are spaces in them?

File: local/test-binary [hunk 2]
```diff
@@ -88,9 +104,8 @@ if [[ "$HOTBACKUP_TESTING" != "no" ]] && [[ -n "${TOKUDB_PLUGIN}" ]] && [[ -n "$
         ${MTR_ARGS} \
         --mysqld-env="LD_PRELOAD=${MYSQLD_ENV}" \
         --suite tokudb.backup \
-        | tee ${WORKDIR_ABS}/tokudb_hotbackup.output \
-        || true
-    awk 'BEGIN { print "<testsuite name=\"Percona Server - TokuDB HotBackup\">" } /^([[:alnum:]]+).*[\[] (retry-)?pass [\]](.*)/ { print "<testcase name=\""$3"\" time=\"" $8/1000 "\"/>" } /^(.*) .*\[ skipped \]/ { print "<testcase name=\""$3"\"><skipped/></testcase>" } /^(.*) .*\[ (retry-)?fail \]/ { print "<testcase name=\""$3"\"><failure/></testcase>" } /^MTR\47s internal check of the test case \47(.*)\47 failed/ { print "<testcase name=\"testcase-check-"$8"\"><failure/></testcase>" } END { print "</testsuite>" }' < ${WORKDIR_ABS}/tokudb_hotbackup.output > ${WORKDIR_ABS}/junit_tokudb_hotbackup.xml
+        | tee ${WORKDIR_ABS}/tokudb_hotbackup.output || true
+    process_mtr_output "Percona Server - TokuDB HotBackup" ${WORKDIR_ABS}/tokudb_hotbackup.output ${WORKDIR_ABS}/junit_tokudb_hotbackup.xml
```
Comment (laurynas-biveinis) [inline local/test-binary]: Do args need quoting in case there are spaces in them?

File: local/test-binary [hunk 3]
```diff
@@ -114,7 +129,7 @@ if [[ "${TOKUDB_ENGINES_MTR}" = "yes" ]] && [[ -n "${TOKUDB_PLUGIN}" ]]; then
             --mysqld=--loose-tokudb_auto_analyze=0 --mysqld=--loose-tokudb_analyze_in_background=false \
             ${TOKUDB_ENGINES_MTR_ARGS} \
             2>&1 | tee ${WORKDIR_ABS}/tokudb.output || true
-    awk 'BEGIN { print "<testsuite name=\"Percona Server - TokuDB\">" } /^([[:alnum:]]+).*[\[] (retry-)?pass [\]](.*)/ { print "<testcase name=\""$3"\" time=\"" $8/1000 "\"/>" } /^(.*) .*\[ skipped \]/ { print "<testcase name=\""$3"\"><skipped/></testcase>" } /^(.*) .*\[ (retry-)?fail \]/ { print "<testcase name=\""$3"\"><failure/></testcase>" } /^MTR\47s internal check of the test case \47(.*)\47 failed/ { print "<testcase name=\"testcase-check-"$8"\"><failure/></testcase>" } END { print "</testsuite>" }' < ${WORKDIR_ABS}/tokudb.output > ${WORKDIR_ABS}/junit_tokudb.xml
+    process_mtr_output "Percona Server - TokuDB" ${WORKDIR_ABS}/tokudb.output ${WORKDIR_ABS}/junit_tokudb.xml
```
Comment (laurynas-biveinis) [inline local/test-binary]: Do args need quoting in case there are spaces in them?

**Review by laurynas-biveinis — COMMENTED** [review]
Looks good in general, minor comments, and needs evidence of testing. Please also change lower versions as applicable

File: local/test-binary (hunk 1 above)
Comment (abychko) [reply local/test-binary]: yes, positional arguments will be split by whitespaces, so "bla bla bla" without quotes will be parsed as 3 arguments, not one

File: local/test-binary (hunk 2 above)
Comment (abychko) [reply local/test-binary]: yes, positional arguments will be split by whitespaces, so "bla bla bla" without quotes will be parsed as 3 arguments, not one

File: local/test-binary (hunk 3 above)
Comment (abychko) [reply local/test-binary]: yes, positional arguments will be split by whitespaces, so "bla bla bla" without quotes will be parsed as 3 arguments, not one

**Comment by abychko** [comment]
test run succeeded https://ps.cd.percona.com/view/8.0/job/test-percona-server-8.0-pipeline/21/

File: local/test-binary (hunk 1 above)
Comment (laurynas-biveinis) [reply local/test-binary]: So it's better to add quotes here, right?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by abychko** [comment]
word `MySQL` doesn't contain whitespaces :)
so, it's not required

**Comment by laurynas-biveinis** [comment]
> word `MySQL` doesn't contain whitespaces :)
> so, it's not required

I am referring to $WORKDIR_ABS in the 2nd and 3rd arg, which currently does not have spaces as it would already fail, but why not make it bulletproof, e.g. for local script testing?

**Comment by abychko** [comment]
yes, you're right.

> bash-4.4$ ./z "Percona Server - TokuDB" "${WORKDIR_ABS}"/tokudb.output "${WORKDIR_ABS}"/junit_tokudb.xml
> Percona Server - TokuDB
> /Users/abychko/tmp/one two three/tokudb.output
> /Users/abychko/tmp/one two three/junit_tokudb.xml

File: local/test-binary [hunk 4]
```diff
@@ -67,9 +84,8 @@ if [[ "${DEFAULT_TESTING}" != "no" ]]; then
         --max-test-fail=0 \
         --suite-timeout=9999 \
         --testcase-timeout=${TESTCASE_TIMEOUT} \
-        | tee ${WORKDIR_ABS}/mtr.output \
-        || status=$?
-    awk 'BEGIN { print "<testsuite name=\"MySQL\">" } /^([[:alnum:]]+).*[\[] (retry-)?pass [\]](.*)/ { print "<testcase name=\""$3"\" time=\"" $8/1000 "\"/>" } /^(.*) .*\[ skipped \]/ { print "<testcase name=\""$3"\"><skipped/></testcase>" } /^(.*) .*\[ (retry-)?fail \]/ { print "<testcase name=\""$3"\"><failure/></testcase>" } /^MTR\47s internal check of the test case \47(.*)\47 failed/ { print "<testcase name=\"testcase-check-"$8"\"><failure/></testcase>" } END { print "</testsuite>" }' < ${WORKDIR_ABS}/mtr.output > $WORKDIR_ABS/junit.xml
+        | tee ${WORKDIR_ABS}/mtr.output || status=$?
+    process_mtr_output MySQL "${WORKDIR_ABS}"/mtr.output "${WORKDIR_ABS}"/junit.xml
```
Comment (laurynas-biveinis) [inline local/test-binary:88]: I'd quote entire args, but it's a stylistic point only

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**